### ISSUE TITLE
prterun SIGSEGV in op_cbfunc while processing PMIx_IOF_dereg call

### ISSUE THREAD

**Issue by drwootton** [openpmix/openpmix#2072]
### What version of the PMIx Reference Library are you using? 
Latest master branch for PMIx and PRRTE

### Describe how PMIx was installed 
Built from github source

### Please describe the system on which you are running

* Operating system/version:
 RHEL 7.6
* Computer hardware:
 Single Power 8 node with 22cores/176 threads
* Network type:
 localhost

-----------------------------

## Details of the problem

I am running prterun with the attach example source where I added a call to PMIx_IOF_dereg immediately preceding the call to PMIx_tool_finalize

First, invoke prterun as **prterun -n 2 --report-uri + hello 30** where prterun displays the URI to connect to and hello waits 30 seconds so I can attach to it.
Next, invoke attach as **attach &LT;prterun-namespace&GT;** where prterun-namepace is the namespace reported in the prterun uri, up to but including the '.'

This fails with a prterun SIGSEGV. Also, the callback in attach that was specified in the PMIx_IOF_dereg call gets called with a UNPACK-PAST-END status, due to the SIGSEGV.

If I run prterun using gdb I get the following traceback at the SIGSEGV
~~~
#0  0x00002000003734ec in op_cbfunc (status=0, cbdata=0x200004100730) at server/pmix_server.c:2846
#1  0x00002000003a7084 in pmix_server_iofdereg (peer=0x200004045d40, buf=0x20000151e478,
    cbfunc=0x2000003732ac <op_cbfunc>, cbdata=0x200004100600) at server/pmix_server_ops.c:3725
#2  0x0000200000388c98 in server_switchyard (peer=0x200004045d40, tag=106, buf=0x20000151e478)
    at server/pmix_server.c:4421
#3  0x00002000003898a4 in pmix_server_message_handler (pr=0x200004045d40, hdr=0x20000407badc,
    buf=0x20000151e478, cbdata=0x0) at server/pmix_server.c:4475
#4  0x000020000047e938 in pmix_ptl_base_process_msg (fd=-1, flags=4, cbdata=0x20000407b9e0)
    at base/ptl_base_sendrecv.c:793
#5  0x000020000066e038 in event_base_loop () from /lib64/libevent_core-2.0.so.5
#6  0x00002000003bf87c in progress_engine (obj=0x101131c8) at runtime/pmix_progress_threads.c:235
#7  0x0000200000768b94 in start_thread () from /lib64/libpthread.so.0
#8  0x00002000008c85f4 in clone () from /lib64/libc.so.6
~~~
It appears that the **cd->peer** value passed in the PMIX_BFROPS_PACK macro is invalid. If I print the structure pointed to by cd->peer, it looks like random data.

The SIGSEGV is due to (cd->peer)->nptr being invalid.

I tried moving the PMIx_IOF_dereg call to immediately precede the attach code which waits for the daemon process to terminate and had the same failure.

This may be a bit intermittent. I did run this test a few times where the only symptom was the UNPACK-PAST-END status.

**Comment by drwootton** [comment]
I updated my code to today's master branch head and I am seeing this same SIGSEGV again. I checked my source and the changes made by #2073 are gone. In backtracking and looking at git blame, I see the #2073 change made last week. But then I see a change made 2 days later that reverses this change back to the original code #2075. (I looked specifically at the first changed line pmix_server_ops.c:3658 to see the changes)
